src/game/objects/ai/trex.c

```diff
@@ -74,6 +76,11 @@ void TRex_Control(int16_t item_num)
     int16_t angle = 0;
 
     if (item->hit_points <= 0) {
+        if (g_Config.disable_trex_collision) {
+            OBJECT_INFO *obj = &g_Objects[item->object_number];
+            obj->collision = NULL;
+        }
```

Comment:
Oh wait, you're right. Consider the following untested approach:

```diff
diff --git a/src/game/objects/ai/trex.c b/src/game/objects/ai/trex.c
index 4ebd4b17..ddbea3d0 100644
--- a/src/game/objects/ai/trex.c
+++ b/src/game/objects/ai/trex.c
@@ -43,7 +43,7 @@ void TRex_Setup(OBJECT_INFO *obj)
     obj->initialise = InitialiseCreature;
     obj->control = TRex_Control;
     obj->draw_routine = DrawUnclippedItem;
-    obj->collision = CreatureCollision;
+    obj->collision = TRex_Collision;
     obj->shadow_size = UNIT_SHADOW / 2;
     obj->hit_points = TREX_HITPOINTS;
     obj->pivot_length = 2000;
@@ -58,6 +58,15 @@ void TRex_Setup(OBJECT_INFO *obj)
     g_AnimBones[obj->bone_index + 44] |= BEB_ROT_Y;
 }
 
+
+void TRex_Collision(int16_t item_num, ITEM_INFO *lara_item, COLL_INFO *coll)
+{
+    if (g_Config.disable_trex_collision && g_Items[item_num].hit_points <= 0) {
+        return;
+    }
+    CreatureCollision(item_num, lara_item, coll);
+}
+
 void TRex_Control(int16_t item_num)
 {
     ITEM_INFO *item = &g_Items[item_num];
diff --git a/src/game/objects/ai/trex.h b/src/game/objects/ai/trex.h
index 43f3b283..1111f09e 100644
--- a/src/game/objects/ai/trex.h
+++ b/src/game/objects/ai/trex.h
@@ -4,4 +4,5 @@
 
 void TRex_Setup(OBJECT_INFO *obj);
 void TRex_Control(int16_t item_num);
+void TRex_Collision(int16_t item_num, ITEM_INFO *lara_item, COLL_INFO *coll);
 void TRex_LaraDeath(ITEM_INFO *item);
```

Reply:
That makes sense, I wasn't sure adding such unique state would be nicer (plus I wasn't sure how to 😆). Will have a look see now.